### PR TITLE
Load UUIDs from S3

### PR DESCRIPTION
**Pull request by peteclark-ft**
Includes:
* Updating circle conf 
* Updating dockerfile to use golang 1.8.3 
* Adding support for loading uuids from S3 
* Some refactoring

### CODE REVIEW

File: native/in_memory_collection.go [hunk 1]
```diff
@@ -24,6 +33,20 @@ func LoadIntoMemory(ctx context.Context, uuidCollection UUIDCollection, collecti
 		return it, nil
 	}
 
+	if skip > 0 && b.s3ReadWriter != nil {
```
Comment (MihaiSorinMoisa) [inline native/in_memory_collection.go]: There are no unit tests for the code inside of this if statement

File: native/in_memory_collection.go [hunk 2]
```diff
@@ -76,6 +99,13 @@ func LoadIntoMemory(ctx context.Context, uuidCollection UUIDCollection, collecti
 		it.append(uuid)
 	}
 
+	if b.s3ReadWriter != nil {
+		err := persistInS3(b.s3ReadWriter, it)
+		if err != nil {
+			log.WithError(err).Warn("Failed to persist collection uuids to bucket")
```
Comment (MihaiSorinMoisa) [inline native/in_memory_collection.go]: There is no unit test that handles this flow

Comment (peteclark-ft) [reply native/in_memory_collection.go]: I'm not sure it's worth testing the logging here - but note we already test the failure cases in the new `s3_test.go` file.

**Comment by peteclark-ft** [comment]
Closing as it's been merged into the Helm Day changes by Bogdan

**Pull request closed without merging**